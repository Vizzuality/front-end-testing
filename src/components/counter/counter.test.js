import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Component from './';

describe('Counter component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Component />).toJSON();
    expect(tree).toMatchSnapshot();
  });

   it('starts with count 0', () => {
     const wrapper = shallow(<Component />);
     const text = wrapper.find('p').text();
     expect(text).toEqual('Current count: 0');
   });

   it('can increment count when increment button is clicked', () => {
      const wrapper = shallow(<Component />);
      const action = wrapper.find('button.increment')

      action.simulate('click');

      const text = wrapper.find('p').text();
      expect(text).toEqual('Current count: 1');
   });

   it('can decrement count when increment button is clicked', () => {
     const wrapper = shallow(<Component />);
     const action = wrapper.find('button.decrement')

     action.simulate('click');

     const text = wrapper.find('p').text();
     expect(text).toEqual('Current count: -1');
  });
});
