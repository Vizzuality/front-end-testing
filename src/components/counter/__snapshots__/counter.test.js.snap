// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches snapshot 1`] = `
<div
  className="counter"
>
  <button
    className="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
  <p
    className="count"
  >
    Current count: 
    0
  </p>
</div>
`;
